### ISSUE TITLE
array-bracket-spacing false positive in array desctructuring syntax

### ISSUE THREAD

**Issue by borekb** [buzinas/tslint-eslint-rules#253]
In our rules:

```
"array-bracket-spacing": [true, "never"]
```

TypeScript's array destructuring syntax:

```
let [ myVar1, myVar2 ] = getVariables();
```

The rule incorrectly thinks that it is an array and prints "There should be no space after "[" (array-bracket-spacing)".

**Comment by buzinas** [comment]
Although you're correct that's not an array, this rule is a mirror of ESLint one, and [as you can see](https://eslint.org/demo/#eyJ0ZXh0IjoibGV0IFsgbXlWYXIxLCBteVZhcjIgXSA9IGdldFZhcmlhYmxlcygpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJhcnJheS1icmFja2V0LXNwYWNpbmciOjJ9LCJlbnYiOnt9fX0=), the same messages shows up when using ESLint.

If you think that's a bug, you should open a PR upstream, and as soon as they fix it, we do it here too.

**Comment by borekb** [comment]
Thanks for the explanation. For reference, this is the canonical ESLint issue: 
Allow different brackets spacing behaviour on deconstructing pattern #7649 https://github.com/eslint/eslint/issues/7649
